adobe_vipm/flows/nav.py

```diff
@@ -1,27 +1,62 @@
+import copy
+import json
 import logging
-from datetime import UTC, datetime
+import os
+from datetime import UTC, datetime, timedelta
 from urllib.parse import urljoin
 
 import requests
 from django.conf import settings
 
 logger = logging.getLogger(__name__)
 
+TOKEN_CACHE_FILE = os.path.join(
+    os.path.expanduser("~"),
+    ".nav-token-cache.json",
+)
+
+
+def get_token_from_disk():
+    if not os.path.exists(TOKEN_CACHE_FILE):
+        return
+
+    token_data = json.load(open(TOKEN_CACHE_FILE))
+    expires_at = datetime.fromisoformat(token_data["expires_at"])
+    if expires_at < datetime.now(UTC):
+        return
+
+    return token_data["access_token"]
+
+
+def save_token_to_disk(token_data):
+    new_token_data = copy.copy(token_data)
+    new_token_data["expires_at"] = (
+        datetime.now(UTC) + timedelta(seconds=token_data["expires_in"] - 300)
+    ).isoformat()
+    with open(TOKEN_CACHE_FILE, "w") as f:
+        f.write(json.dumps(new_token_data))
```

Comment:
`json.dump(f, new_token_data)`

Reply:
🤣  I made this to simplify testing, json dump make several calls to write!